dataset/src/main/kotlin/com/cosmotech/dataset/service/DatasetServiceImpl.kt

```diff
@@ -682,6 +682,12 @@ class DatasetServiceImpl(
       "Number of files must be equal to the number of parts if specified. " +
           "${files.size} != ${datasetCreateRequest.parts?.size}"
     }
+    require(
+        files.groupingBy { it.originalFilename }.eachCount().filter { it.value > 1 }.isEmpty()) {
+          "Part File names should be unique during dataset creation. " +
```

Comment:
It's not obvious to me which type of "part" file this refers to, even with the code so I guess this may be confusing to users.
If I'm not mistaken you are checking the unicity of the `Multipart file names ...` not the `Dataset part names ...`. In addition to the start of the error message, I think the 2 data fields could be made more explicit. Something like `Multipart file names: ...` and `Dataset parts source names: ...`

Same for the update query below.